pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -299,13 +298,28 @@ func (lbaas *LbaasV2) GetLoadBalancer(clusterName string, service *api.Service)
 	return status, true, err
 }
 
+func nodeAddressForLB(node *api.Node) (string, error) {
+	addrs := node.Status.Addresses
+	if len(addrs) == 0 {
+		return "", ErrNoAddressFound
+	}
+
+	for _, addr := range addrs {
+		if addr.Type == api.NodeInternalIP {
```

Comment:
Just to be clear about this: This circuituous behaviour is *exactly* the point of this change.

On OpenStack, we can't rely on having a DNS entry that matches the node name, and we'd rather not have to require k8s node name == OpenStack instance name (but currently we do have to do this).  What we *can* do is reliably detect the instance UUID locally on the instance (ie: in the kubelet), and have this communicated back up through the rest of k8s via node.Status.

So this PR is based on the observation that actually what (presumably) all providers really want *is* the nodes' provider-specific name/ID - and not the k8s node names at all.  So this change passes the full node object instead of node names, allowing the provider LB code to choose which node name/ID they want, in many cases avoiding unnecessary name->ID lookups (and in the OpenStack case, avoiding one source of the k8s node name == openstack instance name requirement).

If we decide we don't want to pass the full node object in the API, another alternative for this change would be to pass the node's provider instance IDs (strings) rather than k8s node names.  Again, the provider *wants* to talk in terms of provider instance IDs because that's exactly what means something to the provider.